jasmine-standalone-2.3.4 (4)/src/Game.js

```diff
@@ -0,0 +1,156 @@
+function Game() {
+  this.frames = [];
+  this.pairs = [];
+}
+
+ // GENERAL METHODS
+
+Game.prototype.isInProgress = function() {
+  return (this._endGameStrike() || this._endGameSpare() || this._lessThan20());
+}
+
+Game.prototype._lastFrame = function() {
+  return this.frames[this.frames.length - 1];
+}
+
+Game.prototype._secondLastFrame = function() {
+  return this.frames[this.frames.length - 2];
+}
+
+Game.prototype._numberOfPairs = function() {
+  return this.pairs.length;
+}
+
+Game.prototype._valueOfPair = function(i) {
+  return Object.values(this.pairs[i])[0];
+}
+
+// ADD FRAME
+
+Game.prototype._addFrame = function() {
+  let framesCount = this.frames.length;
+  let lastFrame = this._lastFrame();
+  let secondLastFrame = this._secondLastFrame();
+
+  if (framesCount === 0 || framesCount === 1) {
+    this.frames.push(1);
+  } else if (lastFrame !== secondLastFrame) {
+    this.frames.push(lastFrame);
+  } else if (lastFrame === secondLastFrame) {
+    if (lastFrame === 10) {
+      this.frames.push(10);
+    } else {
+      this.frames.push(lastFrame + 1);
+    }
+  }
+
+}
+
+// COUNTING METHODS
+
+Game.prototype.basicScore = function() {
+  var counter = 0;
+  for (i = 0; i < this.pairs.length; i ++) {
+   if (this._valueOfPair(i) !== undefined) {
+      if (i < 18) {
```

Comment:
We have a lot of magic numbers here (18, 10, 19 etc). Generally we should avoid this. If we have a constant which represents something we should make it a named constant e.g. MAX_FRAMES = 10 and use that. If we are deriving other numbers from constant we should make it clear how they are derived e.g. (MAX_FRAMES * 2) - 2 and (MAX_FRAMES * 2) - 1. 

Doing this makes it clearer what the intent is. As it stands, it's hard to understand what is happening here.